src/main/kotlin/io/beatmaps/ws/routes/reviews.kt

```diff
@@ -0,0 +1,93 @@
+package io.beatmaps.ws.routes
+
+import io.beatmaps.common.consumeAck
+import io.beatmaps.common.dbo.Review
+import io.beatmaps.common.inlineJackson
+import io.beatmaps.common.rabbitOptional
+import io.ktor.server.routing.Route
+import io.ktor.server.routing.application
+import io.ktor.server.websocket.webSocket
+import kotlinx.datetime.toKotlinInstant
+import kotlinx.serialization.Serializable
+import org.jetbrains.exposed.sql.Op
+import org.jetbrains.exposed.sql.ResultRow
+import org.jetbrains.exposed.sql.and
+import org.jetbrains.exposed.sql.select
+import org.jetbrains.exposed.sql.transactions.transaction
+
+@Serializable
+data class ReviewUpdateInfo(val mapId: Int, val userId: Int)
+
+@Serializable
+data class ReviewWebsocketDTO(
+    val userId: Int,
+    val mapId: Int,
+    val text: String,
+    val sentiment: Int,
+    val createdAt: String,
+    val updatedAt: String,
+    val curatedAt: String?,
+    val deletedAt: String?
+) {
+    companion object {
+        fun wrapRow(row: ResultRow): ReviewWebsocketDTO {
+            return ReviewWebsocketDTO(
+                row[Review.userId].value,
+                row[Review.mapId].value,
+                row[Review.text],
+                row[Review.sentiment],
+                row[Review.createdAt].toKotlinInstant().toString(),
+                row[Review.updatedAt].toKotlinInstant().toString(),
+                row[Review.curatedAt]?.toKotlinInstant()?.toString(),
+                row[Review.deletedAt]?.toKotlinInstant()?.toString()
```

Comment:
Generally if the row has been deleted I'd rather not treat it as still existing. See map deleted where we just return the id. In this case it would have to be the userId and mapId but everything else unset. Maybe a seperate constructor for that?

Reviews especially can be un-deleted by making a new review so you have an edge case where the deleted update actually returns a review that isn't deleted at all.